src/core/tsi/transport_security_adapter.cc

```diff
@@ -148,6 +148,12 @@ static void adapter_destroy(tsi_handshaker* self) {
   gpr_free(self);
 }
 
+static void adapter_shutdown(tsi_handshaker* self) {
```

Comment:
adapter is for sync TSI. You do not need this function. Don't bother to revise. I will delete adapter shortly.